lms/djangoapps/bulk_email/policies.py

```diff
@@ -0,0 +1,20 @@
+
+from edx_ace.policy import Policy, PolicyResult
+from edx_ace.channel import ChannelType
+from opaque_keys.edx.keys import CourseKey
+
+from bulk_email.models import Optout
+
+
+class CourseEmailOptout(Policy):
+
+    def check(self, message):
+        course_id = message.context.get('course_id')
+        if not course_id:
+            return PolicyResult(deny=frozenset())
```

Comment:
Having only a single return seems less readable to me. Matter of taste, I guess.